Buildings/Controls/OBC/ASHRAE/PrimarySystem/BoilerPlant/Controller.mo

```diff
@@ -0,0 +1,2260 @@
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 1/chaIsoValRat
+    "Time to slowly change isolation valve position from fully closed to fully open,
+    should be determined in the field"
+    annotation (Dialog(tab="Staging process parameters", group="Time and delay parameters"));
+
+  parameter Real delPreBoiEna(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 30
+    "Time delay after valve and pump change process has been completed before
+    starting boiler change process"
+    annotation (Dialog(tab="Staging process parameters",group="Time and delay parameters"));
+
+  parameter Real boiChaProOnTim(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 300
+    "Enabled boiler operation time to indicate if it is proven on during a staging
+    process where one boiler is turned on and the other is turned off"
+    annotation (Dialog(tab="Staging process parameters",group="Time and delay parameters"));
+
+  parameter Real delBoiEna(
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 180
+    "Time delay after boiler change process has been completed before turning off
+    excess valves and pumps"
+    annotation (Dialog(tab="Staging process parameters",group="Time and delay parameters"));
+
+  parameter Real k_bypVal(
+    final min=0,
+    final unit="1",
+    displayUnit="1") = 1
+    "Gain of controller"
+    annotation(Dialog(tab="Bypass valve control parameters"));
+
+  parameter Real Ti_bypVal(
+    final min=0,
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 0.5
+    "Time constant of integrator block"
+    annotation(Dialog(tab="Bypass valve control parameters"));
+
+  parameter Real Td_bypVal(
+    final min=0,
+    final unit="s",
+    displayUnit="s",
+    final quantity="time") = 0.1
+    "Time constant of derivative block"
+    annotation(Dialog(tab="Bypass valve control parameters"));
+
+  parameter Real minPumSpePri(
+    final unit="1",
+    displayUnit="1",
+    final min=0,
+    final max=maxPumSpePri) = 0.1
+    "Minimum pump speed"
+    annotation (Dialog(tab="Primary pump control parameters", group="General parameters", enable=have_varPriPum));
+
+  parameter Real maxPumSpePri(
+    final unit="1",
+    displayUnit="1",
+    final min=minPumSpePri,
+    final max=1) = 1
+    "Maximum pump speed"
+    annotation (Dialog(tab="Primary pump control parameters", group="General parameters", enable=have_varPriPum));
+
+  parameter Real VHotWatPri_flow_nominal(
+    final min=1e-6,
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate")
+    "Plant design hot water flow rate thorugh primary loop"
+    annotation (Dialog(group="Boiler plant configuration parameters"));
+
+  parameter Real boiDesFlo[nBoi](
+    final min=1e-6,
+    final unit="m3/s",
+    displayUnit="m3/s",
+    final quantity="VolumeFlowRate")
+    "Vector of design flowrates for all boilers in plant"
```

Comment:
Missing enable condition: only required for primary-only hot water plants with a minimum flow bypass valve, see G36 3.1.8.2.